### ISSUE TITLE
Owncloud shared link not working

### ISSUE THREAD

**Issue by kalletabur** [owncloud/core#16109]
### Steps to reproduce
Let other user share with mail notification .. click link "View it!" (https://my.owncloud.com/index.php/apps/files?dir=%2F&scrollto=Shared%2F20150506.ods).

### Expected behaviour
Clicking the link should take user to this folder/file over browser.

### Actual behaviour
User is sent to owncloud web gui root directory.

### Server configuration
**Operating system**:
Debian 7.8

**Web server:**
NGINX (1.2.1-2.2)

**Database:**
MySQL (5.5.43-0)

**PHP version:**
PHP (5.4.4-14)

**ownCloud version:**
ownCloud 8.0.2-8 (stable) 

**Updated from an older ownCloud or fresh install:**
Updated

**List of activated apps:**
Activity, Deleted files, External storage support, First Run Wizard, LDAP user and group backend, Mail Template Editor, PDF Viewer, Pictures, Server-side Encryption, Share Files, Text Editor, Versions, Video Viewer

**The content of config/config.php:**

```
$CONFIG = array (
  'instanceid' => 'instanceid',
  'passwordsalt' => '',
  'trusted_domains' => 
  array (
    0 => '1.1.1.5',
    1 => 'my.owncloud.com',
  ),
  'datadirectory' => '/var/ownclouddata/',
  'dbtype' => 'mysql',
  'version' => '8.0.2.0',
  'dbname' => 'dbname',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'prefix',
  'dbuser' => 'dbuser',
  'dbpassword' => '',
  'installed' => true,
  'ldapIgnoreNamingRules' => false,
  'forcessl' => true,
  'loglevel' => '0',
  'mail_smtpmode' => 'smtp',
  'mail_smtphost' => 'smtphost',
  'mail_smtpport' => '25',
  'logtimezone' => 'Europe',
  'theme' => '',
  'share_folder' => '/Shared',
  'maintenance' => false,
  'preview_libreoffice_path' => '/usr/bin/libreoffice',
  'secret' => '',
);
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
owncloud

**Are you using encryption:** yes/no
yes

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
AD

#### LDAP configuration (delete this part if not used)

```
|appid|configkey|configvalue|
|user_ldap|cleanUpJobOffset|600|
|user_ldap|enabled|yes|
|user_ldap|has_memberof_filter_support|1|
|user_ldap|home_folder_naming_rule||
|user_ldap|installed_version|0.5.0|
|user_ldap|last_jpegPhoto_lookup|0|
|user_ldap|ldap_agent_password||
|user_ldap|ldap_attributes_for_group_search||
|user_ldap|ldap_attributes_for_user_search||
|user_ldap|ldap_backup_host||
|user_ldap|ldap_backup_port||
|user_ldap|ldap_base|DC=example,DC=local|
|user_ldap|ldap_base_groups|DC=example,DC=local|
|user_ldap|ldap_base_users|DC=example,DC=local|
|user_ldap|ldap_cache_ttl|600|
|user_ldap|ldap_configuration_active|1|
|user_ldap|ldap_display_name|displayname|
|user_ldap|ldap_dn|CN=Comp-Ownc,OU=COM,OU=Company,DC=example,DC=local|
|user_ldap|ldap_email_attr|mail|
|user_ldap|ldap_experienced_admin|0|
|user_ldap|ldap_expert_username_attr|sAMAccountName|
|user_ldap|ldap_expert_uuid_group_attr||
|user_ldap|ldap_expert_uuid_user_attr||
|user_ldap|ldap_group_display_name|cn|
|user_ldap|ldap_group_filter|(&(|(objectclass=group))(|(cn=Dev)(CN=Comp-Ownc)(cn=Comp-admis)))|
|user_ldap|ldap_group_filter_mode|0|
|user_ldap|ldap_group_member_assoc_attribute|member|
|user_ldap|ldap_groupfilter_groups|Comp-DevComp-Owncloud Comp-admis|
|user_ldap|ldap_groupfilter_objectclass|group|
|user_ldap|ldap_host|dc300|
|user_ldap|ldap_login_filter|(&(&(|(objectclass=person))(|(memberof=CN=Comp-Ownc,OU=DL,OU=Company,DC=example,DC=local)(memberof=CN=Comp-Dev,OU=DL,OU=Company,DC=example,DC=local)(memberof=CN=Comp-admis,OU=DL,OU=Company,DC=example,DC=local)))(samaccountname=%uid))|
|user_ldap|ldap_login_filter_mode|0|
|user_ldap|ldap_loginfilter_attributes||
|user_ldap|ldap_loginfilter_email|0|
|user_ldap|ldap_loginfilter_username|1|
|user_ldap|ldap_nested_groups|0|
|user_ldap|ldap_nocase|1|
|user_ldap|ldap_override_main_server|0|
|user_ldap|ldap_paging_size|500|
|user_ldap|ldap_port|389|
|user_ldap|ldap_quota_attr||
|user_ldap|ldap_quota_def||
|user_ldap|ldap_tls|0|
|user_ldap|ldap_turn_off_cert_check|1|
|user_ldap|ldap_user_filter_mode|1|
|user_ldap|ldap_userfilter_groups||
|user_ldap|ldap_userfilter_objectclass||
|user_ldap|ldap_userlist_filter|(&(|(objectclass=person))(|(memberof=CN=Comp-Ownc,OU=DL,OU=Company,DC=example,DC=local)(memberof=CN=Comp-Dev,OU=DL,OU=Company,DC=example,DC=local)(memberof=CN=Comp-SystemAdministrators,OU=DL,OU=Company,DC=example,DC=local)))|
|user_ldap|ocsid|166061|
|user_ldap|types|authentication|

```

### Client configuration
Tested on both op.system

**Browser:**
Any (chrome, ff, opera, IE)

**Operating system:**
Ubuntu 14.04 LTS, Windows 7

#### Web server error log
Access.log
```
10.10.10.5 - - [06/May/2015:12:21:04 +0300] "GET /index.php/apps/files?dir=%2F&scrollto=Shared%2F20150506.ods HTTP/1.1" 200 5160 "-" "Mozilla/5.0 (X11; Linux i686) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/41.0.2272.76 Chrome/41.0.2272.76 Safari/537.36"
10.10.10.5 - - [06/May/2015:12:21:05 +0300] "GET /index.php/core/js/oc.js?v=66bd816353d73de6c398957a8de9dd14 HTTP/1.1" 200 1604 "https://my.owncloud.com/index.php/apps/files?dir=%2F&scrollto=Shared%2F20150506.ods" "Mozilla/5.0 (X11; Linux i686) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/41.0.2272.76 Chrome/41.0.2272.76 Safari/537.36"
10.10.10.5 - - [06/May/2015:12:21:05 +0300] "GET /index.php/avatar/my.username/32?requesttoken=cE8x3sYqm/3AIMcQZk0A6Z.u12L9ak HTTP/1.1" 304 0 "https://my.owncloud.com/index.php/apps/files?dir=%2F&scrollto=Shared%2F20150506.ods" "Mozilla/5.0 (X11; Linux i686) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/41.0.2272.76 Chrome/41.0.2272.76 Safari/537.36"
10.10.10.5 - - [06/May/2015:12:21:07 +0300] "GET /index.php/avatar/my.username/128?requesttoken=cE8x3sYqm/3AIMcQZk0A6Z.u12L9ak HTTP/1.1" 304 0 "https://my.owncloud.com/index.php/apps/files?dir=%2F&scrollto=Shared%2F20150506.ods" "Mozilla/5.0 (X11; Linux i686) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/41.0.2272.76 Chrome/41.0.2272.76 Safari/537.36"
10.10.10.5 - - [06/May/2015:12:21:07 +0300] "GET /cron.php HTTP/1.1" 200 31 "https://my.owncloud.com/index.php/apps/files?dir=%2F&scrollto=Shared%2F20150506.ods" "Mozilla/5.0 (X11; Linux i686) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/41.0.2272.76 Chrome/41.0.2272.76 Safari/537.36"
10.10.10.5 - - [06/May/2015:12:21:07 +0300] "GET /index.php/apps/files_sharing/api/externalShares HTTP/1.1" 200 2 "https://my.owncloud.com/index.php/apps/files?dir=%2F&scrollto=Shared%2F20150506.ods" "Mozilla/5.0 (X11; Linux i686) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/41.0.2272.76 Chrome/41.0.2272.76 Safari/537.36"
10.10.10.5 - - [06/May/2015:12:21:07 +0300] "GET /index.php/apps/files/ajax/list.php?dir=%2F&sort=name&sortdirection=asc HTTP/1.1" 200 3856 "https://my.owncloud.com/index.php/apps/files?dir=%2F&scrollto=Shared%2F20150506.ods" "Mozilla/5.0 (X11; Linux i686) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/41.0.2272.76 Chrome/41.0.2272.76 Safari/537.36"
10.10.10.5 - - [06/May/2015:12:21:07 +0300] "GET /index.php/apps/files/ajax/scan.php?force=false&dir=&requesttoken=cE8x3sYqm/3AIMcQZk0A6Z.u12L9ak HTTP/1.1" 200 96 "https://my.owncloud.com/index.php/apps/files?dir=%2F&scrollto=Shared%2F20150506.ods" "Mozilla/5.0 (X11; Linux i686) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/41.0.2272.76 Chrome/41.0.2272.76 Safari/537.36"
10.10.10.5 - - [06/May/2015:12:21:07 +0300] "GET /index.php/apps/files/ajax/getstoragestats.php?dir=%2F HTTP/1.1" 200 160 "https://my.owncloud.com/index.php/apps/files?dir=%2F&scrollto=Shared%2F20150506.ods" "Mozilla/5.0 (X11; Linux i686) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/41.0.2272.76 Chrome/41.0.2272.76 Safari/537.36"
10.10.10.5 - - [06/May/2015:12:21:08 +0300] "GET /index.php/core/ajax/share.php?fetch=getItemsSharedStatuses&itemType=file HTTP/1.1" 200 625 "https://my.owncloud.com/index.php/apps/files?dir=%2F&scrollto=Shared%2F20150506.ods" "Mozilla/5.0 (X11; Linux i686) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/41.0.2272.76 Chrome/41.0.2272.76 Safari/537.36"

```

No error.log entries

#### ownCloud log (data/owncloud.log)
```
{"reqId":"56c3abbe927770e6d37692575ef37941","remoteAddr":"10.10.10.5","app":"core","message":"OC\\Tags::__construct, tags: Array\n(\n)\n","level":0,"time":"2015-05-06T12:14:07+03:00","method":"GET","url":"\/index.php\/apps\/files\/ajax\/list.php?dir=%2F&sort=name&sortdirection=asc"}


```

**Comment by MorrisJobke** [comment]
Should only be a problem with the link generator

**Comment by MorrisJobke** [comment]
URL generation would be even easier with the permalink feature #11732

**Comment by kalletabur** [comment]
Sorry, title was little misleading .. fixed it

**Comment by MorrisJobke** [comment]
This will also be fixed in 8.0.4
